src/main/drivers/pwm_output_dshot.c

```diff
@@ -144,6 +144,10 @@ void pwmDshotMotorHardwareConfig(const timerHardware_t *timerHardware, uint8_t m
     TIM_TypeDef *timer = timerHardware->tim;
     const IO_t motorIO = IOGetByTag(timerHardware->tag);
 
+    // Boolean configureTimer is always true when different channels of the same timer are processed in sequence,
+    // causing the timer and the associated DMA initialized more than once.
+    // To fix this, getTimerIndex must be expanded to return if a new timer has been requested.
```

Comment:
`getTimerIndex` is very poorly named - it sounds like simple mapping function, but instead it registers timer in table. Splitting it into find and register part would have minimal flash penalty and be cleaner:
```
int timerIndex = getTimerIndex(timer);
if(timerIndex < 0) {
  timerIndex = registerTimer(timer);
  configureTimer = true;
} else {
  configureTimer = false;
}


reconfiguring timer may lead to bugs in future, when `pwmDshotMotorHardwareConfig` is called outside of expected initialization sequence.

Reply:
@martinbudden What do you think?

Btw, the non-HAL code is ready for non-repetitive initialization. The HAL code is not.